src/backend/gl/src/device.rs

```diff
@@ -65,19 +152,108 @@ impl d::Device<B> for Device {
         Ok(n::Heap)
     }
 
-    fn create_renderpass(&mut self, _: &[pass::Attachment], _: &[pass::SubpassDesc], _: &[pass::SubpassDependency]) -> n::RenderPass {
-        unimplemented!()
+    fn create_renderpass(
+        &mut self,
+        attachments: &[pass::Attachment],
+        subpasses: &[pass::SubpassDesc],
+        _dependencies: &[pass::SubpassDependency],
+    ) -> n::RenderPass {
+        let subpasses =
+            subpasses
+                .iter()
+                .map(|subpass| {
+                    let color_attachments =
+                        subpass
+                            .color_attachments
+                            .iter()
+                            .map(|&(index, _)| index)
+                            .collect();
+
+                    n::SubpassDesc {
+                        color_attachments,
+                    }
+                })
+                .collect();
+
+        n::RenderPass {
+            attachments: attachments.into(),
+            subpasses,
+        }
     }
 
     fn create_pipeline_layout(&mut self, _: &[&n::DescriptorSetLayout]) -> n::PipelineLayout {
-        unimplemented!()
+        n::PipelineLayout
     }
 
     fn create_graphics_pipelines<'a>(
         &mut self,
-        _descs: &[(&n::ShaderLib, &n::PipelineLayout, pass::SubPass<'a, B>, &pso::GraphicsPipelineDesc)],
+        descs: &[(&n::ShaderLib, &n::PipelineLayout, pass::Subpass<'a, B>, &pso::GraphicsPipelineDesc)],
     ) -> Vec<Result<n::GraphicsPipeline, pso::CreationError>> {
-        unimplemented!()
+        let gl = &self.share.context;
+        let priv_caps = &self.share.private_caps;
+        descs.iter()
+             .map(|&(shader_lib, layout, ref subpass, desc)| {
+                let subpass = {
```

Comment:
nit: could you just do `let subpass = match subpass.main_pass.subpasses.get(subpass.index) {...}`